design-system/packages/fields/src/DatePicker/components/InputButton.tsx

```diff
@@ -0,0 +1,83 @@
+/** @jsx jsx */
+
+import { ButtonHTMLAttributes, forwardRef } from 'react';
+import { jsx, useTheme } from '@keystone-ui/core';
+import { XIcon } from '@keystone-ui/icons/icons/XIcon';
+import { CalendarIcon } from '@keystone-ui/icons/icons/CalendarIcon';
+import { Adornment, AdornmentWrapper } from './Adornments';
+import { useInputTokens, useInputStyles } from '../..';
+
+type ButtonProps = {
+  invalid?: boolean;
+  isSelected?: boolean;
+  onClear?: () => void;
+} & ButtonHTMLAttributes<HTMLButtonElement>;
+
+export const InputButton = forwardRef<HTMLButtonElement, ButtonProps>(
+  ({ invalid = false, isSelected, onClear, ...props }, ref) => {
+    const { spacing } = useTheme();
+    const inputTokens = useInputTokens({ size: 'medium' });
+    const inputStyles = useInputStyles({ invalid, tokens: inputTokens });
+    const focusStyles = isSelected
+      ? {
+          ...inputStyles[':focus'],
+          ':hover': inputStyles[':focus'],
+          ':focus': inputStyles[':focus'],
+        }
+      : null;
+    const buttonStyles = {
+      ...inputStyles,
+      ...focusStyles,
+      cursor: 'pointer',
+      lineHeight: 'initial', // let the button vertically align its text; the have different native behaviour to inputs
+      textAlign: 'left',
+    } as const;
+
+    return (
+      <AdornmentWrapper shape="square" size="medium">
+        <button aria-invalid={invalid} ref={ref} css={buttonStyles} type="button" {...props} />
+        {onClear && <ClearButton onClick={onClear} />}
+        <Adornment align="right" css={{ paddingRight: spacing.small, pointerEvents: 'none' }}>
+          <CalendarIcon color="dim" />
+        </Adornment>
+      </AdornmentWrapper>
+    );
+  }
+);
+
+const ClearButton = (props: ButtonHTMLAttributes<HTMLButtonElement>) => {
+  const { colors } = useTheme();
+
+  return (
+    <Adornment
+      as="button"
+      align="right"
+      type="button"
```

Comment:
@jossmac not sure if this is something that needs to be fixed in Reckon web DS or not, but when the date-field was ported over to keystone, type="button" attribute was missing, causing unintended submit events to be fired onKeyPress.